tests/unit/structure/GraphCut_unittest.cc

```diff
@@ -168,411 +78,39 @@ TEST(GraphCut, graph_cut_chain)
-		v_factor_type.append_element(ft);
-	}
-}
-
-TEST(GraphCut, graphcuts_sosvm)
-{
-	SGMatrix<int32_t> labels_train;
-	SGMatrix<float64_t> feats_train;
-
-	// Generate random data
-	sg_rand->set_seed(10); // fix the random seed
-	generate_data(10, 40, 40, feats_train, labels_train);
-
-	int32_t num_sample_train  = labels_train.num_cols;
-	int32_t num_classes       = labels_train.num_rows;
-	int32_t dim               = feats_train.num_rows;
-
-	// 1.1 Get edge table
-	SGMatrix< int32_t > edge_table = get_edges_full(num_classes);
-	int32_t num_edges = edge_table.num_rows;
-
-	// 1.2 Define factor type
-	DynArray<CTableFactorType*> v_factor_type;
-	define_factor_types(num_classes, dim, num_edges, v_factor_type);
-
-	// 1.3 Prepare features and labels in factor graph
-	CFactorGraphFeatures* fg_feats_train = new CFactorGraphFeatures(num_sample_train);
-	SG_REF(fg_feats_train);
-	CFactorGraphLabels* fg_labels_train = new CFactorGraphLabels(num_sample_train);
-	SG_REF(fg_labels_train);
-
-	build_factor_graph(feats_train, labels_train, edge_table, v_factor_type, fg_feats_train, fg_labels_train);
-
-	// 1.4 Create factor graph model
-	CFactorGraphModel* model = new CFactorGraphModel(fg_feats_train, fg_labels_train, GRAPH_CUT, false);
-	SG_REF(model);
-
-	// Initialize model parameters
-	for (int32_t u = 0; u < num_classes; u++)
-	{
-		model->add_factor_type(v_factor_type[u]);
-	}
-
-	for (int32_t t = 0; t < num_edges; t++)
-	{
-		model->add_factor_type(v_factor_type[t + num_classes]);
-	}
-
-	// 2.1 Create SGD solver
-	CStochasticSOSVM* sgd = new CStochasticSOSVM(model, fg_labels_train, true);
-	sgd->set_num_iter(150);
-	sgd->set_lambda(0.0001);
-	SG_REF(sgd);
-
-	// 2.2 Train SGD
-	sgd->train();
-
-	// 3.1 Evaluation
-	CStructuredLabels* labels_sgd = CLabelsFactory::to_structured(sgd->apply());
-	SG_REF(labels_sgd);
-	float64_t ave_loss_sgd = 0.0;
-	float64_t acc_loss_sgd = 0.0;
-
-	for (int32_t i = 0; i < num_sample_train; ++i)
-	{
-		CStructuredData* y_pred  = labels_sgd->get_label(i);
-		CStructuredData* y_truth = fg_labels_train->get_label(i);
-		acc_loss_sgd += model->delta_loss(y_truth, y_pred);
-
-		CFactorGraphObservation* y_t = CFactorGraphObservation::obtain_from_generic(y_truth);
-		CFactorGraphObservation* y_p = CFactorGraphObservation::obtain_from_generic(y_pred);
-
-		SGVector<int32_t> s_t = y_t->get_data();
-		SGVector<int32_t> s_p = y_p->get_data();
-
-		// training labels are expected to be correcty predicted for this dataset
-		EXPECT_TRUE(s_t.equals(s_p));
-
-		SG_UNREF(y_pred);
-		SG_UNREF(y_truth);
-	}
-
-	ave_loss_sgd = acc_loss_sgd / static_cast<float64_t>(num_sample_train);
-
-	// training error is expected to be 0
-	EXPECT_EQ(ave_loss_sgd, 0);
+	CFGTestData* fg_test_data = new CFGTestData();
+	SG_REF(fg_test_data);
+	
+	EXPECT_NEAR(fg_test_data->test_sosvm(GRAPH_CUT), 0, 0.1);
 
-	SG_UNREF(labels_sgd);
-	SG_UNREF(sgd);
-	SG_UNREF(model);
-	SG_UNREF(fg_feats_train);
-	SG_UNREF(fg_labels_train);
```

Comment:
If you already did it, then it should be fine ;-) I just wanted to be sure you did check it.